src/libopensc/asn1.c

```diff
@@ -1819,3 +1819,80 @@ sc_encode_oid (struct sc_context *ctx, struct sc_object_id *id,
 	return SC_SUCCESS;
 }
 
+
+#define C_ASN1_SIG_VALUE_SIZE 2
+static struct sc_asn1_entry c_asn1_sig_value[C_ASN1_SIG_VALUE_SIZE] = {
+                { "ECDSA-Sig-Value", SC_ASN1_STRUCT, SC_ASN1_TAG_SEQUENCE | SC_ASN1_CONS, 0, NULL, NULL },
+                { NULL, 0, 0, 0, NULL, NULL }
+};
+
+#define C_ASN1_SIG_VALUE_COEFFICIENTS_SIZE 3
+static struct sc_asn1_entry c_asn1_sig_value_coefficients[C_ASN1_SIG_VALUE_COEFFICIENTS_SIZE] = {
+                { "r", SC_ASN1_OCTET_STRING, SC_ASN1_TAG_INTEGER, SC_ASN1_ALLOC|SC_ASN1_UNSIGNED, NULL, NULL },
+                { "s", SC_ASN1_OCTET_STRING, SC_ASN1_TAG_INTEGER, SC_ASN1_ALLOC|SC_ASN1_UNSIGNED, NULL, NULL },
+                { NULL, 0, 0, 0, NULL, NULL }
+};
+
+
+int
+sc_asn1_sig_value_rs_to_sequence(struct sc_context *ctx, unsigned char *in, size_t inlen,
+                unsigned char **buf, size_t *buflen)
+{
+        struct sc_asn1_entry asn1_sig_value[C_ASN1_SIG_VALUE_SIZE];
+        struct sc_asn1_entry asn1_sig_value_coefficients[C_ASN1_SIG_VALUE_COEFFICIENTS_SIZE];
+	unsigned char *r = in, *s = in + inlen/2;
+	size_t r_len = inlen/2, s_len = inlen/2;
+        int rv;
+
+        LOG_FUNC_CALLED(ctx);
+        sc_copy_asn1_entry(c_asn1_sig_value, asn1_sig_value);
+        sc_format_asn1_entry(asn1_sig_value + 0, asn1_sig_value_coefficients, NULL, 1);
+
+        sc_copy_asn1_entry(c_asn1_sig_value_coefficients, asn1_sig_value_coefficients);
+        sc_format_asn1_entry(asn1_sig_value_coefficients + 0, r, &r_len, 1);
+        sc_format_asn1_entry(asn1_sig_value_coefficients + 1, s, &s_len, 1);
+
+        rv = sc_asn1_encode(ctx, asn1_sig_value, buf, buflen);
+        LOG_TEST_RET(ctx, rv, "ASN.1 encoding ECDSA-SIg-Value failed");
+
+        LOG_FUNC_RETURN(ctx, SC_SUCCESS);
+}
+
+
+int
+sc_asn1_sig_value_sequence_to_rs(struct sc_context *ctx, unsigned char *in, size_t inlen,
```

Comment:
`sc_asn1_sig_value_sequence_to_rs` and `sc_asn1_sig_value_rs_to_sequence` are application specific and look like they belong to the pkcs#15 layer. If moved, you would not need to add them to `libopensc.exports`.